cpp/include/raft/spatial/knn/detail/ivf_pq_search.cuh

```diff
@@ -422,9 +422,8 @@ void postprocess_distances(float* out,        // [n_queries, topk]
         out,
         in,
         len,
-        [scaling_factor] __device__(ScoreT x) -> float {
-          return scaling_factor * scaling_factor * float(x);
-        },
+        raft::scalar_op<raft::binary_compose_op<raft::mul_op, raft::cast_op<float>>, float>(
+          scaling_factor * scaling_factor),
```

Comment:
Just a note on this for reviewers: it's an interesting case because automatic type inference leads to ambiguity (CUDA doesn't know which operator to use to multiply `half` and `float`). So it's a good occasion to try out composition, which arguably doesn't really have advantages here as opposed to passing a custom lambda unless by chance the same composition is used somewhere else in the code. But it does work, perhaps at the expense of readability.

Reply:
What if you do instead unary_compose_op(scalar_mul_op{ scaling_factor * scaling_factor }, raft::cast_op<float>{}). does it compile?

Reply:
Nope, but let me add some constructors, guard with `enable_if` when necessary, and I can make it possible ;)